### ISSUE TITLE
Images are not displayed properly on windows

### ISSUE THREAD

**Issue by zakorgy** [szeged/webrender#68]
https://github.com/szeged/webrender/pull/67 fixed the segfault on windows, but causes a weird behavior when drawing images. The blob example on windows looks like the following:
![blob_wrong](https://user-images.githubusercontent.com/11725775/36421798-2709b6fc-163a-11e8-837d-813389527978.PNG)
The expected result:
![blob_good](https://user-images.githubusercontent.com/11725775/36421956-bbb3ab96-163a-11e8-83b8-b6b1e8117557.PNG)

**Comment by zakorgy** [comment]
With the current master, the blob example doesn't draw anything on windows:
![blob_nothing](https://user-images.githubusercontent.com/11725775/36848151-bb9662e8-1d60-11e8-8293-6d12732bd066.PNG)
Renderdoc shows the following vertex positions:
https://gist.github.com/zakorgy/c63be97d8c41294c591773a917a5a11f
Notice that the `z` vertex coordinates are in the [0, 1] interval in the `gl_PerVertex.gl_Position 2` column.
NOTE: The [near and far plane values ](https://github.com/szeged/webrender/blob/master/webrender/src/internal_types.rs#L63) in our code are -1 and 0, these are -1 and 1 in upstream WebRender.

If we change the near plane to 0 and the far plane to 1, the images are presented:
![blob_wrong](https://user-images.githubusercontent.com/11725775/36421798-2709b6fc-163a-11e8-837d-813389527978.PNG)
Vertex positions: https://gist.github.com/zakorgy/89abd5015d4bbfb78874f281e7d80cdd
The weird thing here is, that the `z` vertex positions are negative values, which in my opinion are incorrect.  (The `x` and `y` vertex coordinates are identical in both case.)
I think we don't create the projections correctly, since we have changed the near and far plane values in our code, also modified the order of the parameters when creating the projections. We can use transformation matrices for Vulkan and D3D12, and keep the original planes and projections, because we still can use them later with OpenGL.

**Comment by dati91** [comment]
I will look into this.

**Comment by dati91** [comment]
So the problem root is [here](https://github.com/szeged/webrender/blob/12ddf211a318c8ba9fefc94db495d0835b56fd6d/webrender/src/device.rs#L657) (as the [first comment](https://github.com/szeged/webrender/issues/68#issue-298562972) states).  The `COLOR_ATTACHMENT` required for drawing into it. But it seems like copying from an image with `COLOR_ATTACHMENT` usage to the render target (which also has the `COLOR_ATTACHMENT` usage) causes this behavior.

**Comment by dati91** [comment]
It seems like a driver bug to me... :/

**Comment by kvark** [comment]
@dati91 it's rarely a driver bug if something behaves differently on different OS/HW in Vulkan. The API is too vague and non-portable. Different drivers get different use of the pipeline barriers, for example. So the incorrect barriers would still work correctly on some platforms. Validation layer helps to an extent...

**Comment by dati91** [comment]
@kvark Yeah, turns out the Intel driver on windows is quite strict, which is (probably) a good thing (or at least it should be, but it is realllly time consuming...).
